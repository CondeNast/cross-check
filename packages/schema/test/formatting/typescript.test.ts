import { subject, setupSchemaTest, teardownSchemaTest} from "../support";

describe("[schema] formatting - typescript", () => {
  beforeEach(setupSchemaTest);
  afterEach(teardownSchemaTest);

  test("simple - published", () => {
    const { format } = subject();
    expect(
      format.published.typescript("SimpleArticle", { name: "SimpleArticle" }),
    ).toMatchSnapshot();
  });

  test("simple - draft", () => {
    const { format } = subject();
    expect(
      format.draft.typescript("SimpleArticle", {
        name: "SimpleArticleDraft"
      }),
    ).toMatchSnapshot();
  });

  test("detailed - published", () => {
    const { format } = subject();
    expect(
      format.published.typescript("MediumArticle", { name: "MediumArticle" }),
    ).toMatchSnapshot();
  });
});