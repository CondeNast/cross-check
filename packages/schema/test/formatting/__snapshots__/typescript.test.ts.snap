// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[schema] formatting - typescript detailed - published 1`] = `
"export interface MediumArticle {
  hed: string;
  dek?: string;
  body: string;
  author?: {
    first?: string;
    last?: string;
  };
  issueDate?: Date;
  canonicalUrl?: URL;
  tags?: Array<string>;
  categories: Array<string>;
  geo?: {
    lat: number;
    long: number;
  };
  contributors?: Array<{
    first?: string;
    last?: string;
  }>;
  relatedArticles?: Array<MediumArticle>;
}"
`;

exports[`[schema] formatting - typescript simple - draft 1`] = `
"export interface SimpleArticleDraft {
  hed?: string;
  dek?: string;
  body?: string;
}"
`;

exports[`[schema] formatting - typescript simple - published 1`] = `
"export interface SimpleArticle {
  hed: string;
  dek?: string;
  body: string;
}"
`;
